x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/oidc/OpenIdConnectRealmSettings.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.security.authc.oidc;
+
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.util.set.Sets;
+import org.elasticsearch.xpack.core.security.authc.RealmSettings;
+import org.elasticsearch.xpack.core.security.authc.support.DelegatedAuthorizationSettings;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+
+
+public class OpenIdConnectRealmSettings {
+
+    private OpenIdConnectRealmSettings() {
+    }
+
+    public static final String TYPE = "oidc";
+
+    public static final Setting.AffixSetting<String> OP_NAME
+        = RealmSettings.simpleString(TYPE, "op.name", Setting.Property.NodeScope);
+    public static final Setting.AffixSetting<String> RP_CLIENT_ID
+        = RealmSettings.simpleString(TYPE, "rp.client_id", Setting.Property.NodeScope);
+    public static final Setting.AffixSetting<SecureString> RP_CLIENT_SECRET
+        = RealmSettings.secureString(TYPE, "rp.client_secret");
+    public static final Setting.AffixSetting<String> RP_REDIRECT_URI
+        = RealmSettings.simpleString(TYPE, "rp.redirect_uri", Setting.Property.NodeScope);
+    public static final Setting.AffixSetting<String> RP_RESPONSE_TYPE
+        = RealmSettings.simpleString(TYPE, "rp.response_type", Setting.Property.NodeScope);
```

Comment:
Maybe we can infer this? Keep it to `code` or `id_token`?

Reply:
You are right that it doesn't make much sense now with the allowed values being `code` and `implicit`. The original thought was to infer "id_token" from the value implicit, but as also proven by me not adding it, this is probably an unnecessary optimization for the user.  I will change `implicit` to `id_token` to be in line with the specification